source/extensions/common/wasm/wasm.cc

```diff
@@ -272,6 +277,30 @@ getCloneFactory(WasmExtension* wasm_extension, Event::Dispatcher& dispatcher,
   };
 }
 
+static WasmEvent toWasmEvent(const std::shared_ptr<WasmHandleBase>& wasm) {
+  if (!wasm) {
+    return WasmEvent::UnableToCreateVM;
+  }
+  switch (wasm->wasm()->fail_state()) {
+  case FailState::Ok:
+    return WasmEvent::Ok;
+  case FailState::UnableToCreateVM:
+    return WasmEvent::UnableToCreateVM;
+  case FailState::UnableToCloneVM:
+    return WasmEvent::UnableToCloneVM;
+  case FailState::MissingFunction:
+    return WasmEvent::MissingFunction;
+  case FailState::UnableToInitializeCode:
+    return WasmEvent::UnableToInitializeCode;
+  case FailState::StartFailed:
+    return WasmEvent::StartFailed;
+  case FailState::ConfigureFailed:
+    return WasmEvent::ConfigureFailed;
+  case FailState::RuntimeError:
+    return WasmEvent::RuntimeError;
+  }
```

Comment:
Nit: add `default` case for older compilers (I think we still need that?)

Reply:
There are other exhaustive cases, e.g. Context::log.  Added NOT_IMPLEMENTED_GCOVR_EXCL_LINE which seems to be the solution to this.

Reply:
`NOT_IMPLEMENTED_GCOVR_EXCL_LINE` should be added as the `default` case, not outside of the `switch`.

Reply:
This is how it is done in context.cc.  This enables the compiler which does correctly support exhaustive cases to report those errors.  Which compiler do you think does not support this?  Given that Envoy is now on c++17, I don't think those compilers are supported any longer.

Reply:
I was thinking about GCC (see: #499). If you want to use it outside of the `default` case, then `NOT_REACHED_GCOVR_EXCL_LINE` is more appropriate than `NOT_IMPLEMENTED_GCOVR_EXCL_LINE`.

I'm assuming that C++ standard doesn't change compiler's behavior, since the codebase is filled with those.

Reply:
Done. PTAL